arroyo-worker/src/operators/aggregating_window.rs

```diff
@@ -215,15 +222,16 @@ impl<K: Key, T: Data, BinA: Data, MemA: Data, OutT: Data>
         }
     }
 
-    async fn handle_watermark(
-        &mut self,
-        _watermark: std::time::SystemTime,
-        ctx: &mut Context<K, OutT>,
-    ) {
-        let Some(watermark) = ctx.watermark() else {return};
-        while self.should_advance(watermark) {
-            self.advance(ctx).await;
+    async fn handle_watermark(&mut self, watermark: Watermark, ctx: &mut Context<K, OutT>) {
+        match watermark {
+            Watermark::EventTime(t) => {
+                while self.should_advance(t) {
+                    self.advance(ctx).await;
+                }
+            }
+            Watermark::Idle => (),
```

Comment:
Could do this with an 
```
if let Watermark::EventTime(t) = watermark {
    while self.should_advance(t) {
        self.advance(ctx).await;
    }
}

Reply:
It's intentionally a match instead of an if let to make it more future-proof. I was worried that if we at some point in the future added another variant (Watermark::ProcessingTime for example) that would introduce bugs in all of the call sites, so I wanted to ensure that would cause a compile error.